src/internal/hooks/use-table-interaction-metrics/index.ts

```diff
@@ -3,8 +3,26 @@
 
 import { useEffect, useRef } from 'react';
 
-import { PerformanceMetrics } from '../../analytics';
+import { ComponentMetrics, PerformanceMetrics } from '../../analytics';
+import { JSONObject } from '../../analytics/interfaces';
 import { useEffectOnUpdate } from '../use-effect-on-update';
+import { useRandomId } from '../use-unique-id';
+
+function useTaskInteractionAttribute(elementRef: React.RefObject<HTMLElement>, value: string) {
```

Comment:
We already have this code in another place:

https://github.com/cloudscape-design/components/blob/56aa09ea8c55009dd7d07f51ef21243da63a29d3/src/internal/hooks/use-performance-marks/index.ts#L13-L27

Can we merge these two into one, to reduce the risk of diverging implementations (e.g. when we make bug fixes)?

Reply:
Fair point. Mine was based off of this one. Will consolidate into a `useDOMAttribute` hook.